src/basic_recipes/voronoiplot.jl

```diff
@@ -0,0 +1,246 @@
+                      smooth=false,
+
+                      # Point settings
+                      markersize=4,
+                      marker=sc.marker,
+                      markercolor=sc.color,
+
+                      # Polygon settings
+                      strokecolor=theme(scene, :patchstrokecolor),
+                      strokewidth=1.0,
+                      color=automatic,
+                      unbounded_edge_extension_factor=0.1,
+                      bounding_box=automatic,
+                      )
+    MakieCore.colormap_attributes!(attr, theme(scene, :colormap))
+    return attr
+end
+
+function clip_polygon(poly::Polygon, edges::Vector{<: Line})
+    # Sutherland-Hodgman algorithm
+    # assuming edges clockwise
+    @assert isempty(poly.interiors) "Polygon must not have holes for clipping."
+
+    # TODO: GeometryBasics has some rotations in this. Do we need that here too?
+    function intersection(A, B, edge)
+        # assuming both infinite
+        e1, e2 = edge
+        alpha = A - e1
+        beta = A - B
+        v = e2 - e1
+        t1, t2 = inv(Mat2f(beta[1], beta[2], v[1], v[2])) * alpha
+        return e1 + t2 * v
+    end
+
+    output = Point2f.(first.(poly.exterior))
+    input = sizehint!(Point2f[], length(output))
+
+    for edge in edges
+        # swap references
+        temp = input
+        input = output
+        output = temp
+
+        empty!(output)
+        v = edge[2] - edge[1]
+        right = Vec2f(v[2], -v[1])
+
+        for i in eachindex(input)
+            p1 = input[mod1(i-1, end)]
+            p2 = input[i]
+            p = intersection(p1, p2, edge)
+            if dot(p2 - edge[1], right) > 0 # p2 inside
+                if dot(p1 - edge[1], right) < 0 # p1 outside
+                    push!(output, p)
+                end
+                push!(output, p2)
+            elseif dot(p1 - edge[1], right) > 0 # p1 inside
+                push!(output, p)
+            end
+        end
+    end
+
+    return Polygon(output)
+end
+
+
+function clip_polygon(poly::Polygon, rect::Rect2)
+    lb = Point2f(minimum(rect))
+    rt = Point2f(maximum(rect))
+    lt = Point2f(lb[1], rt[2])
+    rb = Point2f(rt[1], lb[2])
+    return clip_polygon(poly, [Line(lb, lt), Line(lt, rt), Line(rt, rb), Line(rb, lb)])
+end
+
+function clip_polygon(poly::Polygon, ::Nothing)
+    return poly
+end
+
+function get_voronoi_tiles!(generators, polygons, vorn, bbox)
+    empty!(generators)
+    empty!(polygons)
+    sizehint!(generators, DelTri.num_generators(vorn))
+    sizehint!(polygons, DelTri.num_polygons(vorn))
+    # TODO: get_polygon_coordinates() errors without a bbox tuple.
+    pseudo_bbox = DelTri.polygon_bounds(vorn, 0.1)
+    for i in DelTri.each_generator(vorn)
+        g = DelTri.get_generator(vorn, i)
+        p = Point2f(DelTri.getxy(g))
+        p in bbox && push!(generators, p)
+        polygon_coords = DelTri.get_polygon_coordinates(vorn, i, pseudo_bbox)
```

Comment:
`get_polygon_coordinates()` has some bbox checks in it which make it fail when none is passed, so I'm passing the old auto-generated bbox here. Would be good to avoid this

Reply:
The solution I'm working on will hopefully address this. Do you mean a `nothing` is passed? If you paste here something you'd like to work, I will test against that.

FYI I've already got some existing code for the Sutherland-Hodgan algorithm in my branch on DT.jl for this, don't worry about going further with that.

Reply:
Yes, unfortunately it has to be handled by DT. Since the Voronoi polygons are unbounded, special care is needed for the unbounded edges. The bounding box is necessary so that we can plot them. e.g. if you look at some of the polygon indices, you might see something like

```julia
julia> get_polygon(vorn, 1)
6-element Vector{Int64}:
  6
  2
 -1
 -5
  5
  6
```

these negative numbers (ignore their values, just note that they're negative) denote vertices out at infinity.